### ISSUE TITLE
Makefile is slow even when there's nothing to do

### ISSUE THREAD

**Issue by purpleidea** [purpleidea/mgmt#331]
After 22e4dfa534a25251f10d472eb4e105cfd5728a01 running: `make` takes about six seconds before I see: "Generating: bindata..." where as at: 41eb850b3d72a3d75ef36001d6b0fcdcd7302cc8 it happens instantly.

If you can debug what is so slow, please do and send a fix, otherwise I might have to revert this commit for now.

Thanks,
James

/cc @aequitas

**Comment by aequitas** [comment]
I'm going to chase after this. I'm slowly starting to adopt you habit of prepending time before every command :)

**Comment by purpleidea** [comment]
@aequitas haha, I mostly only do it for demos which I started doing after I added `--converged-timeout` to mgmt, and was using it to show how fast it was.

Thanks!!

**Comment by purpleidea** [comment]
there is one difference in particular (when the 
                    binary is already made) which is that the "Generating: 
                    bindata..." appears instantly in the old version, and in 
                    the new version takes 7 seconds. The issue is *actually* 
                    that it takes 7 seconds to run, when everything is already 
                    built correctly.

So the real bug is that when everything is already good, 
                    it's taking 7 seconds to do nothing.

(or maybe it's doing something unnecessarily)

**Comment by purpleidea** [comment]
Here's some output after make is already run... It should have nothing to do, but it's doing something. So probably not the fault of the latest commit, although that commit does change when the output from make happens.

```
$ time make  --debug=B
GNU Make 4.2.1
Built for x86_64-redhat-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Updating makefiles....
Updating goal targets....
 File 'default' does not exist.
   File 'build' does not exist.
         File 'bindata' does not exist.
        Must remake target 'bindata'.
Generating: bindata...
GNU Make 4.2.1
Built for x86_64-redhat-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Updating makefiles....
Updating goal targets....
 File 'default' does not exist.
   File 'build' does not exist.
  Must remake target 'build'.
  Successfully remade target file 'build'.
Must remake target 'default'.
Successfully remade target file 'default'.
        Successfully remade target file 'bindata'.
         File 'lang' does not exist.
        Must remake target 'lang'.
Generating: lang...
make --quiet -C lang
GNU Make 4.2.1
Built for x86_64-redhat-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Updating makefiles....
Updating goal targets....
 File 'all' does not exist.
   File 'build' does not exist.
  Must remake target 'build'.
  Successfully remade target file 'build'.
Must remake target 'all'.
Successfully remade target file 'all'.
        Successfully remade target file 'lang'.
  Must remake target 'build'.
  Successfully remade target file 'build'.
Must remake target 'default'.
Successfully remade target file 'default'.

real	0m14.377s
user	0m13.925s
sys	0m0.436s

```

**Comment by purpleidea** [comment]
If I switch:

```
GO_FILES := $(shell find . -name '*.go')
```
to:
```
GO_FILES := $(shell echo 'main.go')
```
It's very fast. Even though running:

```
find . -name '*.go'
```

takes:

```
real	0m0.360s
user	0m0.102s
sys	0m0.174s
```

I guess make is getting overwhelmed by all the targets :\

**Comment by purpleidea** [comment]
Fixed in b8b6e5266f91cab4934ee56f307c6e474c5e9027
